actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -910,7 +910,10 @@ def call(env)
 
       def recognize_path(path, environment = {})
         method = (environment[:method] || "GET").to_s.upcase
-        path = Journey::Router::Utils.normalize_path(path) unless path&.include?("://")
+
+        raise TypeError, "path must be a String or Symbol" if path && !(path.is_a?(String) || path.is_a?(Symbol))
```

Comment:
I think if we want to introduce an error for this path, it must be deprecated first.

That is just my feeling though, I haven't looked deeper into the issue.